src/auth/index.js

```diff
@@ -193,6 +193,57 @@ AuthenticationClient.prototype.requestEmailCode = function(data, cb) {
   return this.passwordless.sendEmail(data, cb);
 };
 
+/**
+ * Verify the given OTP which was sent on the given email.
+ *
+ * @method    verifyEmailCode
+ * @memberOf  module:auth.AuthenticationClient.prototype
+ *
+ * @example <caption>
+ *   Given the user credentials (`email` and `otp`), authenticates
+ *   with the provider using the `/oauth/token` endpoint. Upon successful
+ *   authentication, returns a JSON object containing the `access_token` and
+ *   `id_token`.
+ * </caption>
+ *
+ * var data = {
+ *   username: '{EMAIL}'
+ *   otp: '{VERIFICATION_CODE}'
+ * };
+ *
+ * auth0.verifyEmailCode(data, function (err) {
+ *   if (err) {
+ *     // Handle error.
+ *   }
+ * });
+ *
+ * @example <caption>
+ *   The user data object has the following structure.
+ * </caption>
+ *
+ * {
+ *   id_token: String,
+ *   access_token: String,
+ *   token_type: String
+ * }
+ *
+ * @param   {Object}    data              Credentials object.
+ * @param   {String}    data.email        Email.
+ * @param   {String}    data.otp          Verification code.
+ * @param   {Function}  [cb]              Method callback.
+ *
+ * @return  {Promise|undefined}
+ */
+AuthenticationClient.prototype.verifyEmailCode = function(data, cb) {
+  var translatedData = {
+    username: data.email || data.username,
```

Comment:
Let's remove `data.username` and be explicit that you need to provide an email
```suggestion
    username: data.email,
```